flow/multiagent_envs/multiagent_env.py

```diff
@@ -182,36 +172,29 @@ def reset(self):
             self.step_counter = 0
             # issue a random seed to induce randomness into the next rollout
             self.sim_params.seed = random.randint(0, 1e5)
-            # modify the vehicles class to match initial data
-            self.vehicles = deepcopy(self.initial_vehicles)
-            # restart the simulation instance
+            # restart the sumo instance
             self.restart_simulation(self.sim_params)
 
         # perform shuffling (if requested)
         if self.scenario.initial_config.shuffle:
             self.setup_initial_state()
 
         # clear all vehicles from the network and the vehicles class
-        for veh_id in self.traci_connection.vehicle.getIDList():
+        for veh_id in self.k.kernel_api.vehicle.getIDList():  # FIXME: hack
             try:
-                self.traci_connection.vehicle.remove(veh_id)
-                self.traci_connection.vehicle.unsubscribe(veh_id)
-                self.vehicles.remove(veh_id)
+                self.k.vehicle.remove(veh_id)
             except (FatalTraCIError, TraCIException):
                 print("Error during start: {}".format(traceback.format_exc()))
-                pass
 
         # clear all vehicles from the network and the vehicles class
         # FIXME (ev, ak) this is weird and shouldn't be necessary
-        for veh_id in list(self.vehicles.get_ids()):
-            self.vehicles.remove(veh_id)
+        for veh_id in list(self.k.vehicle.get_ids()):
             # do not try to remove the vehicles from the network in the first
             # step after initializing the network, as there will be no vehicles
             if self.step_counter == 0:
                 continue
             try:
-                self.traci_connection.vehicle.remove(veh_id)
-                self.traci_connection.vehicle.unsubscribe(veh_id)
+                self.k.vehicle.remove(veh_id)
```

Comment:
Is the unsubscription moved into the vehicles class?

Reply:
Yup! It does